component/azstorage/datalake_test.go

```diff
@@ -691,6 +692,280 @@ func (s *datalakeTestSuite) TestCreateFile() {
 	s.assert.Empty(props.XMsProperties())
 }
 
+func (s *datalakeTestSuite) TestWriteSmallFile() {
```

Comment:
Can we add some error cases here? Appending to an offset > file size?

Reply:
technically appending offset larger than file size is allowed by the FS it will just be a bunch of `\x00` and we mimic that. I can add a test for this case.

Reply:
Ok so thats a valid test case I see. Do we have it in our UT just to validate that it works and does create 0 data buffers. One part I suspect here is we will create a memory chunk and copy only at certain offset leaving the other byes as is. In lang like C/C++ these untouched bytes will be garbadge and not 0. Can you validate its 0 in go otherwise we will need to fill them explicitly with 0s.

Reply:
Done